### ISSUE TITLE
Update tracker before delete the torrent

### ISSUE THREAD

**Issue by Feriman22** [rakshasa/rtorrent#899]
Hi,

I come from [this thread](https://github.com/Novik/ruTorrent/issues/1922). The author of ruTorrent sent me here.

I would like to set this below:
One torrent reached the defined ratio target and the action is "Remove data (All)". The tracker is not received yet the latest uploaded data size (how much data I uploaded since last refresh), so I have to update tracker before the plugin delete the torrent.

I'm using the [ratio plugin](https://github.com/Novik/ruTorrent/wiki/PluginRatio) of ruTorrent.

Is it possible?

Thank you!

**Comment by pyroscope** [comment]
> The author of ruTorrent sent me here.

Which is a cheap way avoiding an answer.

> ruTorrent use rtorrent implementation of ratio system.

The actions called by the ratio rules are not written by rtorrent however, but by that plugin.

**Comment by Novik** [comment]
> The actions called by the ratio rules

Plugin call combination d.stop/d.close/d.erase. AFAIK this must be enough for updation of corresponding trackers.
In any case, AFAIR, rtorrent xmlrpc API doesn't provide commands for manual update of trackers.
I.e. this is a problem of client.

**Comment by chros73** [comment]
> rtorrent xmlrpc API doesn't provide commands for manual update of trackers.

Not just that but this kind of behaviour should be included in the client itself.
There's a similar issue when [updating tracker is also needed when leeching a download has been finished](https://github.com/rakshasa/libtorrent/issues/167#issuecomment-379489632).

I wouldn't await for a quick fix though, there are pending PRs that will modify the current behaviour of finishing downloading.

**Comment by rakshasa** [comment]
Yes, this is an issue with the client not guaranteeing the event is sent to the tracker afaics.

Not a priority as I'm redoing other parts, however PRs are welcome.

**Comment by Feriman22** [comment]
Thank you all! I'm waiting for the fix.

**Comment by chros73** [comment]
> however PRs are welcome

If the rest of my patches will be merged then I'll probably take a quick look into this ;)
